### PR TITLE
Add new risks

### LINKED ISSUE #126: SitRep and Cards changes
https://docs.google.com/spreadsheets/d/15pBV7lLXxhPUYLFL0sB9f8coERp48C5VSCRpJ4B1ark/edit#gid=0

Don't touch the mission itself yet, just implement the sitreps and have them in the random pool for starters. Mission changes will be in a different PR

Note that we will need new CA risks for the negative sitreps

### PR DESCRIPTION
**Pull request by statusNone**
closes #126

### CODE REVIEW

File: CovertInfiltration/Src/CovertInfiltration/Classes/X2StrategyElement_InfiltrationMissionSources.uc [hunk 1]
```diff
@@ -217,3 +220,46 @@ static function Phase2OnSuccess(XComGameState NewGameState, XComGameState_Missio
...
+				`CI_Log("ADDING TO LIST");
+				ActiveSitReps.AddItem(Risk.RiskTemplateName);
+			}
+		}
+	}
+
+	if(ActiveSitReps.Length > 0)
+	{
+		`CI_Log("DIDN'T PROC");
+		return ActiveSitReps;
```
Comment (Xymanek) [inline CovertInfiltration/Src/CovertInfiltration/Classes/X2StrategyElement_InfiltrationMissionSources.uc]: Do this in any case, whether it procced or not. No need for code below

**Comment by Xymanek** [comment]
> ok so i can either:
a) change the name of the risk templates (which breaks the naming convention of covert action risk templates)
b) create a struct to link the name of the risk to the name of the sit rep
c) create a subclass of `X2CovertActionRiskTemplate` and add variable to link the risk to the sitrep

B is better - A is too constrained, C is too complicated

File: CovertInfiltration/Src/CovertInfiltration/Classes/X2StrategyElement_InfiltrationActions.uc [hunk 2]
```diff
@@ -6,7 +6,7 @@
 //  WOTCStrategyOverhaul Team
 //---------------------------------------------------------------------------------------
 
-class X2StrategyElement_InfiltrationActions extends X2StrategyElement;
+class X2StrategyElement_InfiltrationActions extends X2StrategyElement config(GameData);
```
Comment (Xymanek) [inline CovertInfiltration/Src/CovertInfiltration/Classes/X2StrategyElement_InfiltrationActions.uc]: Why config?

File: CovertInfiltration/Src/CovertInfiltration/Classes/X2StrategyElement_InfiltrationActions.uc [hunk 3]
```diff
@@ -25,86 +25,72 @@ static function array<X2DataTemplate> CreateTemplates()
...
+
+	`CREATE_X2TEMPLATE(class'X2CovertInfiltrationTemplate', Template, CovertActionName);
+
+	foreach class'X2Helper_Infiltration'.default.FlatRiskSitReps(FlatRiskSitRep)
+	{
+		Template.Risks.AddItem(FlatRiskSitRep.FlatRiskName);
+
+		if (bCreateSlots)
+		{
+			Template.Slots.AddItem(CreateDefaultStaffSlot('InfiltrationStaffSlot'));
```
Comment (Xymanek) [inline CovertInfiltration/Src/CovertInfiltration/Classes/X2StrategyElement_InfiltrationActions.uc]: so... 6x5=30 slots?

File: CovertInfiltration/Src/CovertInfiltration/Classes/X2StrategyElement_InfiltrationActions.uc [hunk 4]
```diff
@@ -25,86 +25,72 @@ static function array<X2DataTemplate> CreateTemplates()
 	return CovertActions;
 }
 
+static function X2DataTemplate CreateInfiltrationTemplate(name CovertActionName, optional bool bCreateSlots=false)
```
Comment (Xymanek) [inline CovertInfiltration/Src/CovertInfiltration/Classes/X2StrategyElement_InfiltrationActions.uc]:
```
static function X2CovertInfiltrationTemplate CreateInfiltrationTemplate
```
And remove the casts later

File: CovertInfiltration/Src/CovertInfiltration/Classes/X2StrategyElement_InfiltrationMissionSources.uc [hunk 5]
```diff
@@ -222,44 +222,40 @@ static function Phase2OnSuccess(XComGameState NewGameState, XComGameState_Missio
...
+	local ActionFlatRiskSitRep FlatRiskSitRep;
 	local array<name> ActiveSitReps;
 	local CovertActionRisk Risk;
 
+	InfiltrationState = XComGameState_MissionSiteInfiltration(MissionState);
+
 	foreach `XCOMHISTORY.IterateByClassType(class'XComGameState_CovertAction', ActionState)
 	{
-		if (ActionState.Location == MissionState.Location)
+		if (ActionState.GetReference() == InfiltrationState.CorrespondingActionRef)
```
Comment (Xymanek) [inline CovertInfiltration/Src/CovertInfiltration/Classes/X2StrategyElement_InfiltrationMissionSources.uc]: Just fetch the action by id, why iterate?

File: CovertInfiltration/Src/CovertInfiltration/Classes/X2StrategyElement_InfiltrationActions.uc (hunk 3 above)
Comment (statusNone) [reply CovertInfiltration/Src/CovertInfiltration/Classes/X2StrategyElement_InfiltrationActions.uc]: WAT

**Pull request merged**